### DISCUSSION TITLE
Search within playlist

### DISCUSSION THREAD

**Discussion by firhatop** [pear-devs/pear-desktop#2327] [Ideas]
Adding a search feature within the playlist would be great.

**Comment by ArjixWasTaken** [comment]
## Assuming you are talking about the playlist queue, on the right side of the player, and not the playlist page


hmmm, I can think of a few possible ways of implementing that, each with a varying level of difficulty

1. entirely re-implement the playlist player, so we can add custom functionality to it
2. introduce a search modal that is independent of the playlist UI
3. inject a search bar to the playlist list, and only search among the already present songs

2 and 3 are probably the easiest ways of doing it, but also kinda the most lacking feature-wise

2 is probably both the easiest and most practical

**Reply by ArjixWasTaken** [reply]
personally I am in favour of number 1, but admittedly it is the hardest
